File: TechAcadStudentZMVC/TechAcadStudentZMVC/Controllers/HomeController.vb
Public Class HomeController
    Inherits System.Web.Mvc.Controller

    Function Index() As ActionResult
        Return View()
    End Function

    Function About() As ActionResult
        ViewData("Message") = "Your application description page."

        Return View()
    End Function

    Function Contact() As ActionResult
        ViewData("Message") = "Contact Page - The Tech Academy."

        Return View()
    End Function

    Function Instructors() As ActionResult
        Return View()
    End Function

    Function Instructor(int id) As ActionResult
        ViewBag.Id = id
        Return View()
    End Function


End Class

File: TechAcadMVC/TechAcadMVC/Controllers/HomeController.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Web.Mvc;
using TechAcadMVC.Models;

namespace TechAcadMVC.Controllers
{
    public class HomeController : Controller
    {
        public ActionResult Index()
        {
            return View();
        }

        public ActionResult About()
        {
            ViewBag.Message = "Your application description page.";

            return View();
        }

        public ActionResult Contact()
        {
            ViewBag.Message = "Contact Page - The Tech Academy.";

            return View();
        }
        public ActionResult Instructors()
        {
            List<Instructor> instructors = new List<Instructor>
            {
                new Instructor
                {
                    Id = 1,
                    FirstName = "James",
                    LastName = "Marcello"
                },
                new Instructor
                {
                    Id = 2,
                    FirstName = "Brett",
                    LastName = "Chandler"
                },
                new Instructor
                {
                    Id = 3,
                    FirstName = "Adam",
                    LastName = "Smithosian"
                }
            };
            return View(instructors);
        }
        public ActionResult Instructor(int id)
        {
            ViewBag.Id = id;
            Instructor dayTimeInstructor = new Instructor

            {
                Id = 1,
                FirstName = "Nick",
                LastName = "Walker"
            };

            return View(dayTimeInstructor);
        }
    }
}